### PR TITLE
Add DataFrame.pivot_table.

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR adds the method `DataFrame.pivot_table` to enhance pandas API compatibility. It uses the exact same arguments as `cudf.pivot_table` but automatically supplies the first argument (a DataFrame).

Related: #11314

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/tests/test_reshape.py [hunk 1]
```diff
@@ -596,6 +596,42 @@ def test_pivot_table_simple(data, aggfunc, fill_value):
...
+            "D": np.random.randn(24),
+            "E": np.random.randn(24),
+        }
+    ],
+)
+@pytest.mark.parametrize(
+    "aggfunc", ["mean", "count", {"D": "sum", "E": "count"}]
+)
+@pytest.mark.parametrize("fill_value", [0])
+def test_dataframe_pivot_table_simple(data, aggfunc, fill_value):
```
Comment (mroeschke) [inline python/cudf/cudf/tests/test_reshape.py:615]: What the difference between this test and `test_pivot_table_simple`? I can't immediately tell

**Comment by mroeschke** [comment]
Should all the arguments that aren't supported and raise `NotImplementedError` be tested as well? https://github.com/rapidsai/cudf/pull/11314/files#diff-a3d6b9fab46a4b2a1de7c903eab52bda471251b40452b1ac172d9d02a35a8f2bR1341

File: python/cudf/cudf/tests/test_reshape.py (hunk 1 above)
Comment (bdice) [reply python/cudf/cudf/tests/test_reshape.py:615]: The previous test uses `cudf.pivot_table`, and the new test uses `DataFrame.pivot_table`. That is the only difference. I am open to alternatives here. I initially decided against a parametrization of the previous test but could be swayed to do whatever you think is right.

**Comment by bdice** [comment]
> Should all the arguments that aren't supported and raise `NotImplementedError` be tested as well? https://github.com/rapidsai/cudf/pull/11314/files#diff-a3d6b9fab46a4b2a1de7c903eab52bda471251b40452b1ac172d9d02a35a8f2bR1341

Since this code is a pure wrapper around the module-level function, I didn’t add any tests beyond some API parametrizations. We could do that for completeness but code coverage won’t change.

File: python/cudf/cudf/tests/test_reshape.py (hunk 1 above)
Comment (mroeschke) [reply python/cudf/cudf/tests/test_reshape.py:615]: Ah I see it now, thanks. Yeah I don't think it's worth parameterizing here.

**Comment by mroeschke** [comment]
> Since this code is a pure wrapper around the module-level function, I didn’t add any tests beyond some API parametrizations. We could do that for completeness but code coverage won’t change.

SGTM

**Review by mroeschke — APPROVED** [review]

**Review by shwina — APPROVED** [review]

**Comment by shwina** [comment]
@gpucibot merge

**Pull request merged**